test/libsolutil/JSON.cpp

```diff
@@ -135,179 +135,202 @@ BOOST_AUTO_TEST_CASE(parse_json_strict)
 	// Trailing garbage is not allowed in ECMA-262
 	BOOST_CHECK(!jsonParseStrict("{\"1\":2,\"2\":\"2\",\"3\":{\"3.1\":\"3.1\",\"3.2\":3}}}}}}}}}}", json, &errors));
 
-	// Comments are not allowed in ECMA-262
-	// ... but JSONCPP allows them
```

Comment:
And we allowed comments in inputs as well before the switch? That may actually be a real problem then...

Reply:
Worst thing that could happen is that we may need to preprocess away comments, which'd be annoying, but fine, since luckily those comments aren't within strings, so things like source locations will still be fine - but let's chat and check if this is actually needed first.

Reply:
See https://github.com/ethereum/solidity/pull/14877#discussion_r1516027130